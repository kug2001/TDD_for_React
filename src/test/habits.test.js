import React from 'react';
import { screen, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import Habits from '../components/habits';

describe('habits에 대한 테스트 코드', () => {
  const habits = [
    { id: 1, name: 'study', count: 1 },
    { id: 2, name: 'sleep', count: 0 },
  ]
  let onIncrement;
  let onDecrement;
  let onDelete;
  let onAdd;
  let onReset;
  beforeEach(() => {
    onIncrement = jest.fn();
    onDecrement = jest.fn();
    onDelete = jest.fn();
    onAdd = jest.fn();
    onReset = jest.fn();
    render(<Habits
      habits={habits}
      onIncrement={onIncrement}
      onDecrement={onDecrement}
      onDelete={onDelete}
      onAdd={onAdd}
      onReset={onReset}
    />)
  })

  it('스탭샷 테스트 코드', () => {
    const component = renderer.create(
      <Habits
        habits={habits}
        onIncrement={onIncrement}
        onDecrement={onDecrement}
        onDelete={onDelete}
        onAdd={onAdd}
        onReset={onReset}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  describe('habits에 대한 테스트 코드', () => {
    it('리셋버튼이 눌러졌을 때 onReset이 호출', () => {
      const handleReset = screen.getByText('Reset All');
      userEvent.click(handleReset);
      expect(onReset).toHaveBeenCalledTimes(1);
    });
    it('리셋버튼이 눌러졌을 때 onReset이 호출', () => {
      const handleReset = screen.getByText('Reset All');
      userEvent.click(handleReset);
      expect(onReset).toHaveBeenCalledTimes(1);
    });
  })
})