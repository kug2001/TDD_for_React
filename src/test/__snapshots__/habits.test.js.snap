// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`habits에 대한 테스트 코드 스탭샷 테스트 코드 1`] = `
<div
  className="habits"
>
  <form
    className="add-form"
    onSubmit={[Function]}
  >
    <input
      className="add-input"
      placeholder="Habit"
      type="text"
    />
    <button
      className="add-button"
    >
      Add
    </button>
  </form>
  <ul>
    <li
      className="habit"
    >
      <span
        className="habit-name"
      >
        study
      </span>
      <span
        className="habit-count"
        data-testid="count"
      >
        1
      </span>
      <button
        className="habit-button habit-increase"
        data-testid="increase"
        onClick={[Function]}
      >
        <i
          className="fas fa-plus-square"
        />
      </button>
      <button
        className="habit-button habit-decrease"
        data-testid="decrease"
        onClick={[Function]}
      >
        <i
          className="fas fa-minus-square"
        />
      </button>
      <button
        className="habit-button habit-delete"
        data-testid="delete"
        onClick={[Function]}
      >
        <i
          className="fas fa-trash"
        />
      </button>
    </li>
    <li
      className="habit"
    >
      <span
        className="habit-name"
      >
        sleep
      </span>
      <span
        className="habit-count"
        data-testid="count"
      >
        0
      </span>
      <button
        className="habit-button habit-increase"
        data-testid="increase"
        onClick={[Function]}
      >
        <i
          className="fas fa-plus-square"
        />
      </button>
      <button
        className="habit-button habit-decrease"
        data-testid="decrease"
        onClick={[Function]}
      >
        <i
          className="fas fa-minus-square"
        />
      </button>
      <button
        className="habit-button habit-delete"
        data-testid="delete"
        onClick={[Function]}
      >
        <i
          className="fas fa-trash"
        />
      </button>
    </li>
  </ul>
  <button
    className="habits-reset"
    onClick={[MockFunction]}
  >
    Reset All
  </button>
</div>
`;
